platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -173,16 +173,14 @@ export function ViewportGridProvider({ children, service }) {
             displaySetOptions,
             viewportLabel: viewportLabels[viewportIndex],
           };
+          newViewport.viewportOptions.viewportId ||= newViewport.viewportId || `viewport-${viewportIndex}`;
```

Comment:
Under what circumstances does the || part come in to play? For `SET_DISPLAYSETS_FOR_VIEWPORTS` the viewports on screen are potentially getting a new display set, but don't they already have an id since they are on screen?

Reply:
It is possible to specify the viewport options specifically, which may or may not include a new viewport id, so one should be generated if not already present.

Reply:
See my response to line 275 comment.